RiotSwiftUI/Modules/Room/AllChatsOnboarding/View/AllChatsOnboarding.swift

```diff
@@ -0,0 +1,92 @@
+// 
+// Copyright 2021 New Vector Ltd
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+import SwiftUI
+
+struct AllChatsOnboarding: View {
+
+    // MARK: - Properties
+    
+    // MARK: Private
+    
+    @Environment(\.theme) private var theme: ThemeSwiftUI
+    
+    // MARK: Public
+    
+    @ObservedObject var viewModel: AllChatsOnboardingViewModel.Context
+    
+    var body: some View {
+        VStack {
+            Text(VectorL10n.allChatsOnboardingTitle)
+                .font(theme.fonts.title3SB)
+                .foregroundColor(theme.colors.primaryContent)
+                .padding()
+            TabView {
+                ForEach(viewModel.viewState.pages) { page in
+                    pageView(image: page.image,
+                             title: page.title,
+                             message: page.message)
+                }
+            }
+            .tabViewStyle(PageTabViewStyle())
+
+            Button { viewModel.send(viewAction: .cancel) } label: {
+                Text(VectorL10n.allChatsOnboardingTryIt)
+            }
+            .buttonStyle(PrimaryActionButtonStyle())
+            .padding()
+        }
+        .background(theme.colors.background.ignoresSafeArea())
+        .frame(maxHeight: .infinity)
+        .onAppear {
+            self.setupAppearance()
+        }
+    }
+    
+    @ViewBuilder
+    private func pageView(image: UIImage, title: String, message: String) -> some View {
+        VStack {
+            Spacer()
+            Image(uiImage: image)
+            Spacer()
+            Text(title)
+                .font(theme.fonts.title2B)
+                .foregroundColor(theme.colors.primaryContent)
+                .padding(.bottom, 16)
+            Text(message)
+                .multilineTextAlignment(.center)
+                .font(theme.fonts.callout)
+                .foregroundColor(theme.colors.primaryContent)
+            Spacer()
+        }
+        .padding(.horizontal)
+    }
+    
+    private func setupAppearance() {
+        let tintColor: UIColor = theme.isDark ? .white : .black
+        UIPageControl.appearance().currentPageIndicatorTintColor = tintColor
+        UIPageControl.appearance().pageIndicatorTintColor = tintColor.withAlphaComponent(0.2)
```

Comment:
Does this affect page controls elsewhere in the app? Might be my misunderstanding - I've always thought the static `.appearance()` is a global theme.

Reply:
@pixlwave 

It does but swiftUI forces the `UIPageControl` to white all the time. I just force the colour to the default one. I've checked and there is no other way to do as off now ....

Reply:
@pixlwave 
I can use `.indexViewStyle(.page(backgroundDisplayMode: .always))` which is the official way to do but it will look like this:

| Light Mode | Dark Mode |
| - | - |
| ![Simulator Screen Shot - iPhone 12 Pro - 2022-08-23 at 15 30 39](https://user-images.githubusercontent.com/15386762/186170946-53add3c3-63bc-492a-a97d-66bdee3300ee.png) | ![Simulator Screen Shot - iPhone 12 Pro - 2022-08-23 at 15 30 25](https://user-images.githubusercontent.com/15386762/186171013-e71cbc95-2946-47da-8282-5ac92f370828.png) |

Is it ok for you @niquewoodhouse ?

Reply:
No need to change it from my perspective if it doesn't affect elsewhere in the app, just wanted to check if that was the case :)